extension/goext/sync.go

```diff
@@ -47,3 +47,13 @@ type ISync interface {
 	// Update updates a path with given json
 	Update(ctx context.Context, path string, json string) error
 }
+
+type ErrCompacted struct {
```

Comment:
reviewers, I've put this type in goext to avoid converting back and forth from internal types (eg. sync.ErrCompacted) to external types (eg. goext.ErrCompacted).
In fact the problem is more general, we already have duplication of some types, eg. DbOptions, (log) Level, LockPolicies etc..

Note that goext pakacge depends only on the standard library (and pkg/errors), so I started to think that goext may the best package to put definition of common interfaces!

Reply:
Actually this is the only place where you can put any struct/interface which should be shared with Go extension. I wonder if this restriction applies since we began using Go modules.

Reply:
>  I wonder if this restriction applies since we began using Go modules.

Even if it doesn't it's still a good thing to have a clear distinction on what's private and what's not. If we actually start allowing extensions to use anything from outside of `goext` then we will have a much harder time modifying anything here.

Reply:
I know, I'm just curious ;)